lib/improver/tests/convection/test_diagnoseconvectiveprecipitation.py

```diff
@@ -0,0 +1,378 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the convection.DiagnoseConvectivePrecipitation plugin."""
+
+
+import unittest
+
+
+from cf_units import Unit
+import iris
+from iris.coords import AuxCoord, DimCoord
+from iris.cube import Cube
+from iris.tests import IrisTest
+import numpy as np
+
+from improver.convection import DiagnoseConvectivePrecipitation
+
+
+mm_hr_to_m_s = 2.7778e-7
+
+
+def set_up_precipitation_rate_cube():
+    """Create a cube with metadata and values suitable for
+    precipitation rate."""
+    data = np.zeros((1, 1, 4, 4))
+    # Convert from mm/hr to m/s.
+    data[0, 0, 0, :] = 2.0 * mm_hr_to_m_s
+    data[0, 0, 1, :] = 4.0 * mm_hr_to_m_s
+    data[0, 0, 2, :] = 8.0 * mm_hr_to_m_s
+    data[0, 0, 3, :] = 16.0 * mm_hr_to_m_s
+    data[0, 0, 0, 2] = 0.0
+    data[0, 0, 2, 1] = 0.0
+    data[0, 0, 3, 0] = 0.0
+    return set_up_cube(data, "lwe_precipitation_rate", "m s-1")
+
+
+def set_up_cube(data, phenomenon_standard_name, phenomenon_units,
```

Comment:
I haven't done this, as although I did originally copy this from the `ensemble_calibration` helper functions, I've also made some edits. Therefore making this change might involve quite a bit of fiddling with `ensemble_calibration` unit tests. I hope that the work to sort this out gets picked up as part of #50.